pinot-connectors/pinot-flink-connector/src/main/java/org/apache/pinot/connector/flink/sink/PinotSinkFunction.java

```diff
@@ -151,4 +148,18 @@ private void flush()
       LOG.info("Pinot segment uploaded to {}", segmentURI);
     });
   }
+
+  @Override
+  public List<GenericRow> snapshotState(long checkpointId, long timestamp) {
+    return _pendingRows;
+  }
+
+  @Override
+  public void restoreState(List<GenericRow> pendingRows)
+      throws Exception {
+    for (GenericRow row: pendingRows) {
+      _segmentWriter.collect(row);
+      _segmentNumRecord++;
+    }
+  }
```

Comment:
need to put pendingRows back into _pendingRows. otherwise in the case of immediate failure after restore, and a snapshot is triggered. those data a lost.